finality-aleph/src/party/mod.rs

```diff
@@ -470,7 +482,52 @@ where
         };
         loop {
             let last_finalized_number = self.client.info().finalized_number;
-            debug!(target: "afa", "Highest finalized: {:?} session {:?}", last_finalized_number, session_id);
+            if last_finalized_number > first_block {
+                let next_session_id = SessionId(session_id.0 + 1);
+                let next_session_authorities = match self
+                    .client
+                    .runtime_api()
+                    .next_session_authorities(&BlockId::Number(first_block))
+                {
+                    Ok(authorities) => authorities
+                        .expect("authorities for next session must be available at first block of current session"),
+                    Err(e) => {
+                        error!(target: "afa", "Error when getting authorities for session {:?} {:?}", next_session_id, e);
+                        return;
```

Comment:
This deserves a panic. This`return` can actually lead to unexpected behaviour

Reply:
The idea is that, we `return` from the session if and only if the session is finalized (last block of it)

Reply:
Done